### ISSUE TITLE
Ability to skip face detection and pass already prepared image.

### ISSUE THREAD

**Issue by HrachMD** [serengil/deepface#313]
Hi @serengil ! Thanks a lot for nice implementation. 
I would ask you for an enhancement. Please, make possible to omit the face detection phase. In case if I have already cropped and aligned ROI of the face, it will be time saving to skip that stages in your framework's flow. I've found in params only align=Bool option, but will be nice to able to set for ex. detector_backed=None to skip the processing. Thanks!

**Comment by serengil** [comment]
Sure, I can do it in the next release. But I'm not sure I can do it soon.

**Comment by HrachMD** [comment]
Thanks. Great to have it for the next release. 
Currently I'm using tiny yolo model, trained by me. It's works pretty accurate and fast compared with SSD. It detects not only frontal faces, but also from lateral view and other perspectives. I can share the weights with you.

**Comment by serengil** [comment]
mtcnn and retinaface are my favorite. They can handle not only frontal images and they can align faces well.

Still, I appreciate to know the details of your model.

**Comment by HrachMD** [comment]
> mtcnn and retinaface are my favorite. They can handle not only frontal images and they can align faces well.
> 
> Still, I appreciate to know the details of your model.

Sure. Find out them here - https://www.dropbox.com/sh/y5a0u48j1q29ozu/AADGRN7BiC_-HavlEsYU5W8sa?dl=0

It is very lightweight but the accuracy still preserved. Detects faces at any angle and has 4 classes - face, eye, nose, lips.

**Comment by HrachMD** [comment]
So, if you like my model, I will be glad to make my contribution and write a wrapper for your detectors. I continuously works on my model performance improvements. The live example how it works on production can be find out here: https://apps.apple.com/am/app/imed-cloud-fast-edit/id1499218144

**Comment by 12bitmisfit** [comment]
you can do this with some hackery... Most of the code below is code from deepface but with the facial detection stripped out.

```
from deepface.basemodels import VGGFace, OpenFace, Facenet, FbDeepface
from deepface.commons import functions, distance
from tensorflow.keras.preprocessing import image
import numpy as np

# Loading Module, can be VGGFace, OpenFace, Facenet, FbDeepface
model = OpenFace.loadModel()
target_size_0, target_size_1 = functions.find_input_shape(model)

# Make face image fit required size by the network
img = your cropped / preprocessed face image/array
factor0 = target_size_0 / img.shape[0]
factor1 = target_size_1 / img.shape[1]
factor = min(factor0, factor1)
dsize = (int(img.shape[1] * factor), int(img.shape[0] * factor))
img = cv2.resize(img, dsize)
diff0 = target_size_0 - img.shape[0]
diff1 = target_size_1 - img.shape[1]
img_pixels = np.pad(img, ((diff0 // 2, diff0 - diff0 // 2), (diff1 // 2, diff1 - diff1 // 2), (0, 0)), 'constant')
img_pixels = image.img_to_array(img_pixels)
img_pixels /= 255

# Compute Facial Encoding
encoding = model.predict(image)[0]

# You can put all of that into a function then store those facial encodings in an array or a dictionary with names, if you wanted to use deepface's distance functions to compare facial encodings you could do something like the following

# Lets assume you have two facial encodings, face1 & face2

dst = distance.findCosineDistance(face1, face2)
or
dst = distance.findEuclideanDistance(face1, face2)
```

**Comment by serengil** [comment]
I just pushed an update to the code repo. You can pull the latest code to have the fixed code. I'm also going to publish it on pip in the next couple of days.

you should set "skip" to the detector_backend to have this ability
